account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/client/SignerClient.java

```diff
@@ -0,0 +1,196 @@
+package com.alchemy.aa.client;
+
+import com.alchemy.aa.Stamper;
+import com.alchemy.aa.Stamper.Stamp;
+import com.alchemy.aa.client.api.AuthJWT.Request;
+import com.alchemy.aa.client.api.AuthJWT.Response;
+import com.alchemy.aa.client.api.AuthUser;
+import com.alchemy.aa.client.api.AuthUser.TurnKeyWhoAmIRequest;
+import com.alchemy.aa.client.api.AuthUser.WhoAmIRequest;
+import com.alchemy.aa.client.api.SignRawMessage.SignParameters;
+import com.alchemy.aa.client.api.SignRawMessage.SignRawMessageRequest;
+import com.alchemy.aa.client.api.SignRawMessage.SignedResponse;
+import com.alchemy.aa.client.api.SignRawMessage.SigningBody;
+import com.alchemy.aa.client.api.StampedRequest;
+import com.alchemy.aa.core.CredentialBundle;
+import com.alchemy.aa.core.TekManager;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.google.crypto.tink.util.Bytes;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import lombok.Builder;
+import lombok.Getter;
+import org.bouncycastle.util.encoders.Hex;
+
+public class SignerClient {
+
+    @Builder
+    public record User(String email, String userId, String orgId, String address, String solanaAddress) {
+    }
+
+    @Getter
+    private enum PathName {
+        LOOKUP("lookup"), AUTH("auth"), AUTH_JWT("auth-jwt"), WHOAMI("whoami"), SIGN_PAYLOAD("sign-payload");
+
+        private final String name;
+
+        PathName(String name) {
+            this.name = name;
+        }
+    }
+
+    private final HttpConfig httpConfig;
+    private final ObjectMapper mapper;
+    private final Logger logger;
+
+    public SignerClient(HttpConfig httpConfig) {
+        this.httpConfig = httpConfig;
+        this.logger = Logger.getLogger(getClass().getName());
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * Inject a bunlde to stamper and authenticate the with orgId.
+     *
+     * @param tekManager
+     *            client's tek keys.
+     * @param orgId
+     * @param bundle
+     *            bundle from alchemy signer service.
+     *
+     * @return authenticated user
+     *
+     * @throws Exception
+     */
+    public Stamper authenticateWithBundle(TekManager tekManager, String orgId, String bundle) throws Exception {
```

Comment:
Question: does it make sense to make the `SignerClient` have a `TekManager` as a dependency rather than passing in the `TekManager` every time? Do we expect this method to be called on the same client with different TekManagers?